libs/api/ebpf_api.cpp

```diff
@@ -478,9 +478,13 @@ ebpf_map_update_element(fd_t map_fd, _In_ const void* key, _In_ const void* valu
     if ((type == BPF_MAP_TYPE_PROG_ARRAY) || (type == BPF_MAP_TYPE_HASH_OF_MAPS) ||
         (type == BPF_MAP_TYPE_ARRAY_OF_MAPS)) {
         fd_t fd = *(fd_t*)value;
-        ebpf_handle_t handle = _get_handle_from_file_descriptor(fd);
-        if (handle == ebpf_handle_invalid) {
-            return EBPF_INVALID_FD;
+        ebpf_handle_t handle = ebpf_handle_invalid;
+        // If fd is ebpf_fd_invalid, this call is to clear the fd from the map.
```

Comment:
Comment doesn't match code?

Reply:
The comment means that if the fd passed to the API is `ebpf_fd_invalid`, then this call is to clear the entry, hence we need to pass `ebpf_handle_invalid` to the IOCTL call.

Reply:
OK, but the comment says "if the fd is ebpf_fd_invalid", but the check is the opposite:
```
if (fd != ebpf_fd_invalid) {
```

Maybe make it clearer: "If this is a valid fd, resolve to a handle, otherwise pass ebpf_handle_invalid to the IOCTL.

Reply:
Fixed.